RHDEVDOCS-4445 - Added 1.6.1, 1.5.6 and 1.4.12 release notes (#50294)
openshift/openshift-docs: modules/gitops-release-notes-1-4-12.adoc

// Module included in the following assembly:
//
// * gitops/gitops-release-notes.adoc

:_content-type: REFERENCE

[id="gitops-release-notes-1-4-12_{context}"]
= Release notes for {gitops-title} 1.4.12

{gitops-title} 1.4.12 is now available on {product-title} 4.8, 4.9, and 4.10.

[id="fixed-issues-1-4-12_{context}"]
== Fixed issues

The following issues have been resolved in the current release:

* Before this update, in a large set of application the application controllers were restarted multiple times due to the unresponsiveness of liveness probes. This update fixes the issue by removing the liveness probe in the application controller `statefulset`.  link:https://issues.redhat.com/browse/GITOPS-2153[GITOPS-2153]

* Before this update, the certificate of RHSSO cannot be validated when it is setup with a certificate which is not signed by certificate authorities. This update fixes the issue and now the users can provide a custom certificate which will be used in verifying the Keycloak's TLS certificate when communicating with it. link:https://issues.redhat.com/browse/GITOPS-2214[GITOPS-2214]

> 

Reviewer: please include the details and instructions outlined in the  Release notes text of  https://issues.redhat.com/browse/GITOPS-2214

* Before this update, a terminating Argo CD managed namespace blocked the creation of roles and other configuration of other managed namespaces. This update fixes this issue. link:https://issues.redhat.com/browse/GITOPS-2277[GITOPS-2277]

> * Before this update, the dex pod in the multicloud-gitops-hub namespace did not appear. This update fixes this issue. link:https://issues.redhat.com/browse/GITOPS-2235[GITOPS-2235] 

Reviewer: ```suggestion
* Before this update, Dex pods failed to start with CreateContainerConfigError when a SCC of 'anyuid' is assigned to the Dex ServiceAccount. This update fixes this issue by assigning a default user id to the Dex container.
  link:https://issues.redhat.com/browse/GITOPS-2235[GITOPS-2235] 
```

// Module included in the following assembly:
//
// * gitops/gitops-release-notes.adoc

:_content-type: REFERENCE

[id="gitops-release-notes-1-4-12_{context}"]
= Release notes for {gitops-title} 1.4.12

{gitops-title} 1.4.12 is now available on {product-title} 4.8, 4.9, and 4.10.

[id="fixed-issues-1-4-12_{context}"]
== Fixed issues

> The following issues have been resolved in the current release:

Reviewer 2: Personally, I am happy with the passive voice here.

But if you want to avoid it, as per the style guide, you could say:

"The current release resolves the following issues:"

Author: @gabriel-rh keeping the passive voice, since all the previous release notes have the same.

> * Before this update, in a large set of application the application controllers were restarted multiple times due to the unresponsiveness of liveness probes. This update fixes the issue by removing the liveness probe in the application controller `statefulset`.  link:https://issues.redhat.com/browse/GITOPS-2153[GITOPS-2153]

Reviewer 2: ```suggestion
* Before this update, in a large set of applications, the application controllers were restarted multiple times due to the unresponsiveness of liveness probes. This update fixes the issue by removing the liveness probe in the application controller `StatefulSet`.  link:https://issues.redhat.com/browse/GITOPS-2153[GITOPS-2153]
```

> * Before this update, the certificate of RHSSO cannot be validated when it is setup with a certificate which is not signed by certificate authorities. This update fixes the issue and now you can provide a custom certificate which will be used in verifying the Keycloak's TLS certificate when communicating with it. You can add the `rootCA` to their Argo CD custom resource `.spec.keycloak.rootCA` field. The operator reconciles to this change and updates the `oidc.config` in `argocd-cm` configmap with the PEM encoded root certificate. 

Reviewer 2: ```suggestion
* Before this update, the RHSSO certificate cannot be validated when it is set up with a certificate which is not signed by certificate authorities. This update fixes the issue and now you can provide a custom certificate which will be used in verifying the Keycloak's TLS certificate when communicating with it. You can add the `rootCA` to the Argo CD custom resource `.spec.keycloak.rootCA` field. The Operator reconciles this change and updates the `oidc.config` field in the `argocd-cm` `ConfigMap` with the PEM-encoded root certificate. 
```

+
[NOTE]
====

> Restart the Argo CD server pod after updating the `.spec.keycloak.rootCA`.

Reviewer 2: ```suggestion
Restart the Argo CD server pod after updating the `.spec.keycloak.rootCA` field.
```

====
+
For example: 

[source,yaml]
----
apiVersion: argoproj.io/v1alpha1
kind: ArgoCD
metadata:
  name: example-argocd
  labels:
    example: basic
spec:
  sso:
    provider: keycloak
    keycloak:
     rootCA: |
       ---- BEGIN CERTIFICATE ----
       This is a dummy certificate
       Please place this section with appropriate rootCA
       ---- END CERTIFICATE ----
  server:
    route:
      enabled: true
----
+
link:https://issues.redhat.com/browse/GITOPS-2214[GITOPS-2214]

> * Before this update, a terminating Argo CD managed namespace blocked the creation of roles and other configuration of other managed namespaces. This update fixes this issue. link:https://issues.redhat.com/browse/GITOPS-2277[GITOPS-2277]

Reviewer 2: ```suggestion
* Before this update, a terminating namespace that was managed by Argo CD would block the creation of roles and other configuration of other managed namespaces. This update fixes this issue. link:https://issues.redhat.com/browse/GITOPS-2277[GITOPS-2277]
```

> * Before this update, the Dex pods failed to start with `CreateContainerConfigError` when an SCC of `anyuid` was assigned to the Dex ServiceAccount. This update fixes this issue by assigning a default user id to the Dex container. link:https://issues.redhat.com/browse/GITOPS-2235[GITOPS-2235] 

Reviewer 2: ```suggestion
* Before this update, the Dex pods failed to start with `CreateContainerConfigError` when an SCC of `anyuid` was assigned to the Dex `ServiceAccount` resource. This update fixes this issue by assigning a default user id to the Dex container. link:https://issues.redhat.com/browse/GITOPS-2235[GITOPS-2235] 
```
